docs/developer-docs/setup/pocket-ic.md

```diff
@@ -0,0 +1,278 @@
+# Testing canisters with PocketIC
+
+## Overview
+
+PocketIC is a lightweight, deterministic testing platform for programmatic testing for canisters written in Rust and Python. It seamlessly integrates with existing testing infrastructure, such as `cargo test`, and runs as a standalone binary that doesn't require additional containers or virtual machines. Currently, PocketIC is supported on MacOS and Linux host systems. 
+
+While there are other options for testing canisters, they may come with pitfalls such as:
+
+- Installing and testing canisters on the mainnet provides the 'real' workflow experience, but will cost developers real cycles.
+
+- Testing using the local replica provided by `dfx` allows for testing on a single IC node, but there is no cross-net or multi-subnet functionality. Testing with the local replica is not deterministic, and it is rather heavyweight.
+
+PocketIC is designed to remedy these pitfalls, resulting in a testing environment that provides the following benefits:
+
+- Deterministic: Synchronous control over the local execution environment.
```

Comment:
Since it exposes a REST API, interacting with PocketIC in a synchronous manner will entirely depend on the language that's used to write the test. Maybe here is the right place to mention that PocketIC removes non-deterministic parts of the replica to make tests fully reproducible.